Sources/ComposableArchitecture/SharedState/References/ValueReference.swift

```diff
@@ -120,7 +177,13 @@ final class ValueReference<Value>: Reference, @unchecked Sendable {
         initialValue: initialValue
       ) { [weak self] value in
         guard let self else { return }
-        self.lock.withLock { self._value = value ?? initialValue }
+        #if canImport(Perception)
+          self._$perceptionRegistrar.willSet(self, keyPath: \.value)
+          defer { self._$perceptionRegistrar.didSet(self, keyPath: \.value) }
+        #endif
```

Comment:
this is a fix that is independent of this PR. I also added a failing test that this fixes.